File: CS 320 Enhancement/ContactService/ContactService.py
#Programmer: Hannah Frederick
#Date:09/20/2025
#Program Description: Utilize in-memory data structures to store, edit, and delete contact, task, and appointment information for a mobile application. Utilize Pytest unit testing to verify that all requirements are working as intended

import pytest
import sys
import os
import ipytest
ipytest.autoconfig()

from Contact import Contact

class ContactService:

    #Create a list and set the currentId to 0 so we have a starting ID
    def __init__(self):
        self.contact_list = []
        self.current_id = 0

    #adds a contact and assigns it an Id
    def add_contact(self, first_name, last_name, phone_number,address):
        string_id = str(self.current_id)
        
        new_contact = Contact(string_id, first_name, last_name, phone_number, address)
        self.contact_list.append(new_contact)

        self.current_id = self.current_id + 1

    #Deletes a contact
    def delete_contact(self, contact_id):
        for contact in self.contact_list:
            if contact.get_contact_id() == contact_id:
                self.contact_list.remove(contact)
                return True
        return False

    #Updates the first name
    def update_first_name(self, contact_id, first_name):
        for contact in self.contact_list:
            if contact.get_contact_id() == contact_id:
                contact.set_first_name(first_name)
                return True
        return False

    #Updates the last name
    def update_last_name(self, contact_id, last_name):
        for contact in self.contact_list:
            if contact.get_contact_id() == contact_id:
                contact.set_last_name(last_name)
                return True
        return False

    #Updates the phone number
    def update_phone_number(self, contact_id, phone_number):
        for contact in self.contact_list:
            if contact.get_contact_id() == contact_id:
                contact.set_phone_number(phone_number)
                return True
        return False

    #Updates the address
    def update_address(self, contact_id, address):
        for contact in self.contact_list:
            if contact.get_contact_id() == contact_id:
                contact.set_address(address)
                return True
        return False

File: CS 320 Project Original/CS 320 Project Original/CS 320 Project/src/Contact_Service/ContactService.java
package Contact_Service;

import java.util.ArrayList;

public class ContactService {
	
	//Create an array and set the currentId to 0 so we have a starting ID
	public static  ArrayList<Contact> contactArray = new ArrayList<Contact>();
	int currentId = 0;
	
	//adds a contact and assigns it an Id
	public void addContact(String firstName, String lastName, String phoneNumber, String address) {
		
		String stringId = Integer.toString(currentId);
		
		Contact newContact = new Contact(stringId, firstName, lastName, phoneNumber, address);
		contactArray.add(newContact);
		
		currentId = currentId + 1;
		
	}
	
	
	//Deletes a contact
	public void deleteContact(String contactId) {

		for( int i = 0; i < contactArray.size(); i++) {
			if (contactArray.get(i).getContactId().equals(contactId)) {
				contactArray.remove(i);
				break;
			}
		
		}
	}
	
	//Updates the first name
	public void updateFirstName(String contactId, String firstName) {
		
		for( int i = 0; i < contactArray.size(); i++) {
			if (contactArray.get(i).getContactId().equals(contactId)) {
				contactArray.get(i).setFirstName(firstName);
				break;
			}
		}
	}
	
	//Updates the last name
	public void updateLastName(String contactId, String lastName) {
		
		for( int i = 0; i < contactArray.size(); i++) {
			if (contactArray.get(i).getContactId().equals(contactId)) {
				contactArray.get(i).setLastName(lastName);
				break;
			}
		}
	}
	
	//Updates the phone number
	public void updatePhoneNumber(String contactId, String phoneNumber) {
		
		for( int i = 0; i < contactArray.size(); i++) {
			if (contactArray.get(i).getContactId().equals(contactId)) {
				contactArray.get(i).setPhoneNumber(phoneNumber);
				break;
			}
		}
	}
	
	//Updates the address
	public void updateAddress(String contactId, String address) {
		
		for( int i = 0; i < contactArray.size(); i++) {
			if (contactArray.get(i).getContactId().equals(contactId)) {
				contactArray.get(i).setAddress(address);
				break;
			}
		}
	}
	
}